### PR TITLE
🚀 Speedup and cleanup "path" testcases and benchmarks

### PR DESCRIPTION
**Pull request by ReneWerner87**
i have tried to reduce the benchmark cases for routing matching 
and move the testcases in a list, so that all test and benchmark functions use the same lists instead of repeating them redundantly

### CODE REVIEW

**Comment by ReneWerner87** [comment]
image
instead of
image

File: path_testcases_test.go [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+			{
+				pattern: "/api/v1/:param<int>?",
+				testCases: []routeTestCase{
+					{url: "/api/v1/entity", params: nil, match: false},
+					{url: "/api/v1/8728382", params: []string{"8728382"}, match: true},
+					{url: "/api/v1/true", params: nil, match: false},
+					{url: "/api/v1/", params: []string{""}, match: true},
+				},
+			},
+		}...,
```
Comment (gaby) [inline path_testcases_test.go:716]: Are these golang dots or spaces? 😂

Comment (efectn) [reply path_testcases_test.go:716]: I think it's to necessary make it variadic

Comment (ReneWerner87) [reply path_testcases_test.go:716]: golang dots for the variadic function -> for the slice combination

**Comment by ReneWerner87** [comment]
do you have a better name for my helper file ? 
path_testcases_test.go

File: path_testcases_test.go (hunk 1 above)
Comment (gaby) [reply path_testcases_test.go:716]: I was looking at it on my phone, and got confused :-)

**Comment by ReneWerner87** [comment]
image
lines before and after in the test file

**Comment by ReneWerner87** [comment]
definitely need an approval flag if you are good with the pull request 😄

**Review by gaby — APPROVED** [review]
Lgtm, easier to read!

**Pull request merged**